common/bytes.go

```diff
@@ -61,9 +62,10 @@ func HasHexPrefix(str string) bool {
 	return l >= 2 && str[0:2] == "0x"
 }
 
+var hexRegex = regexp.MustCompile(`^0[xX]([0-9a-fA-F]{2})+$`)
+
 func IsHex(str string) bool {
-	l := len(str)
-	return l >= 4 && l%2 == 0 && str[0:2] == "0x"
+	return hexRegex.MatchString(str)
 }
```

Comment:
The only use of IsHex is in IsHexAddress, also in package common. I think it would be best to delete IsHex and improve IsHexAddress instead. Bonus points if you can do it without a regular expression ;)